GCC 16 build error for RE_DUP_MAX in minrx.cpp

```
make rxgrep

g++ -std=c++20 -O3 -Wall   -c -o minrx.o minrx.cpp
minrx.cpp: In member function ‘MinRX::Compile::Subexp MinRX::Compile::mkrep(const Subexp&, MinRX::NInt, MinRX::NInt,MinRX::NInt)’:
minrx.cpp:906:44: error: ‘RE_DUP_MAX’ was not declared in this scope; did you mean ‘RAND_MAX’?
  906 |                 if ((m != (NInt) -1 && m > RE_DUP_MAX) || (n != (NInt) -1 && n > RE_DUP_MAX) || m > n)
      |                                            ^~~~~~~~~~
      |                                            RAND_MAX
make: *** [<builtin>: minrx.o] Error 1
```

Need to define `_POSIX_C_SOURCE=200809L` to pull in `RE_DUP_MAX` from `limits.h` via C++ `<climits>`

Sending a PR now

--- a/Makefile
+++ b/Makefile
@@ -16,8 +16,9 @@ builds/$(TYPE)/meson-info:
 	meson setup builds/$(TYPE) --prefix=$(PREFIX) --buildtype=$(TYPE)
 
 # traditional build (requires only make): make tryit
-CFLAGS=-O3 -Wall
-CXXFLAGS=-std=c++20 -O3 -Wall
+CFLAGS=-O3 -Wall -D_POSIX_C_SOURCE=200809L
+CXXFLAGS=-std=c++20 -O3 -Wall -D_POSIX_C_SOURCE=200809L
+
 rxgrep: minrx.o rxgrep.o
 	$(CXX) -o $@ $^
 tryit: minrx.o tryit.o

